base/src/AudioToTextXForm.cpp

```diff
@@ -0,0 +1,212 @@
+#include "AudioToTextXForm.h"
+#include "FrameMetadata.h"
+#include "FrameMetadataFactory.h"
+#include "Frame.h"
+#include "Logger.h"
+#include "Utils.h"
+#include "whisper.h"
+#include "SFML/Config.hpp"
+
+AudioToTextXFormProps::AudioToTextXFormProps(
+	DecoderSamplingStrategy _samplingStrategy,
+	std::string _modelPath,
+	int _bufferSize) : samplingStrategy(_samplingStrategy),
+	modelPath(_modelPath),
+	bufferSize(_bufferSize)
+{}
+
+size_t AudioToTextXFormProps::getSerializeSize() {
+	return ModuleProps::getSerializeSize() +
+		sizeof(samplingStrategy) +
+		sizeof(modelPath) +
+		sizeof(bufferSize);
+}
+
+template <class Archive>
+void AudioToTextXFormProps::serialize(Archive& ar, const unsigned int version) {
+	ar& boost::serialization::base_object<ModuleProps>(*this);
+	ar& samplingStrategy;
+	ar& modelPath;
+	ar& bufferSize;
+}
+
+class AudioToTextXForm::Detail
+{
+public:
+	Detail(AudioToTextXFormProps& _props) : mProps(_props)
+	{
+	}
+	~Detail() {}
+
+	whisper_full_params fetchDefaultParams() {
+		auto samplingStrategy = whisper_sampling_strategy::WHISPER_SAMPLING_GREEDY;
+		switch (mProps.samplingStrategy)
+		{
+		case AudioToTextXFormProps::DecoderSamplingStrategy::GREEDY:
+			samplingStrategy = whisper_sampling_strategy::WHISPER_SAMPLING_GREEDY;
+			break;
+		case AudioToTextXFormProps::DecoderSamplingStrategy::BEAM_SEARCH:
+			samplingStrategy = whisper_sampling_strategy::WHISPER_SAMPLING_BEAM_SEARCH;
+			break;
+		default:
+			throw AIPException(AIP_FATAL, "Unknown Sampling Strategy");
+		}
+		return whisper_full_default_params(samplingStrategy);
+	}
+
+	void setProps(AudioToTextXFormProps& props)
+	{
+		mProps = props;
+	}
+
+public:
+	framemetadata_sp mOutputMetadata;
+	std::string mOutputPinId;
+	std::vector<float> inputAudioBuffer;
+	AudioToTextXFormProps mProps;
+	int mFrameType;
+	whisper_context *mWhisperContext = NULL;
```

Comment:
We Should Initialize mWhisperContext in Constructor

Reply:
It is initialized in init of pipeline.

Reply:
Yeah, but it's a  good idea that default value should be set in Constructor, instead of setting while declaring